### ISSUE TITLE
Update of Broker from 8.3.5 to 8.4.0 fails

### ISSUE THREAD

**Issue by michaelholtermann** [camunda/zeebe#16159]
**Describe the bug**
We're running Zeebe Broker 8.3.5 (and Gateway and the Hazelcast Exporter and Zeebe Simple Monitor) on our Kubernetes Cluster.

When Upgrading to 8.4.0, the follower partitions do not become live, the cluster remains unhealthy. Downgrading again to 8.3.5 solved the issue.

The result looks like this, screenshot from Zeebe Simple Monitor (we named the Zeebe Broker "process-control-tower-broker"):
![image](https://github.com/camunda/zeebe/assets/83571016/15ea76c4-1dc3-4377-9b0a-0a3df8a92f3b)

Apparently, the followers on the same node cannot be started.

The logs below refer to `broker-1`.

I can reproduce this issue, so I'd be happy to provide more details if required. For now, I'll stay with 8.3.5.

Any help is appreciated.

**To Reproduce**
Update an existing 8.3.5 Broker, running on a K8s Stateful Set on AWS EKS, to 8.4.0.

**Expected behavior**

Update runs smoothly as the ones before.

**Log/Stacktrace**

I see several exceptions in the logs, some `WARN`s as well as some `ERROR`s:
<details><summary>CompletionException</summary>
<p>
Probably a normal one, not critical:

```
2024-01-30 14:23:22.927 [Broker-1] [zb-actors-1] [] WARN 
      io.camunda.zeebe.topology.gossip.ClusterTopologyGossiper - Failed to sync with 2
java.util.concurrent.CompletionException: io.atomix.cluster.messaging.MessagingException$RemoteHandlerFailure: Remote handler failed to handle message, cause: Failed to handle message, host process-control-tower-broker-1.process-control-tower-broker.services.svc.cluster.local:26502 is not a known cluster member
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$25(NettyMessagingService.java:626) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31) ~[guava-33.0.0-jre.jar:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$26(NettyMessagingService.java:624) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
	at io.atomix.cluster.messaging.impl.AbstractClientConnection.dispatch(AbstractClientConnection.java:49) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.atomix.cluster.messaging.impl.AbstractClientConnection.dispatch(AbstractClientConnection.java:30) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.atomix.cluster.messaging.impl.NettyMessagingService$MessageDispatcher.channelRead0(NettyMessagingService.java:1109) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.104.Final.jar:4.1.104.Final]
	at java.base/java.lang.Thread.run(Unknown Source) ~[?:?]
```

</p>
</details>
followed by:

<details><summary> java.lang.IndexOutOfBoundsException </summary>
 <p>

A similar exception is thrown for `partition 1` as well:

```
2024-01-30 14:23:25.925 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.roles.FollowerRole - RaftServer{raft-partition-partition-3}{role=FOLLOWER} - Accepted PollRequest{term=134, candidate=2, lastLogIndex=151925082, lastLogTerm=134}: candidate's log is up-to-date
2024-01-30 14:23:25.944 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.roles.FollowerRole - RaftServer{raft-partition-partition-3}{role=FOLLOWER} - Accepted VoteRequest{term=135, candidate=2, lastLogIndex=151925082, lastLogTerm=134}: candidate's log is up-to-date
2024-01-30 14:23:25.956 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-3} - Found leader 2
2024-01-30 14:23:25.960 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-3} - Setting firstCommitIndex to 0. RaftServer is ready only after it has committed events upto this index
2024-01-30 14:23:25.968 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-3} - Setting firstCommitIndex to 0. RaftServer is ready only after it has committed events upto this index
2024-01-30 14:23:26.206 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-3} - Setting firstCommitIndex to 151925083. RaftServer is ready only after it has committed events upto this index
2024-01-30 14:23:26.206 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-3} - Commit index is 151925083. RaftServer is ready
2024-01-30 14:23:26.206 [Broker-1] [raft-server-1-3] [raft-server-3] INFO 
      io.atomix.raft.partition.impl.RaftPartitionServer - RaftPartitionServer{raft-partition-partition-3} - Server successfully bootstrapped partition PartitionId{id=3, group=raft-partition} in 2732ms
2024-01-30 14:23:26.210 [Broker-1] [zb-actors-0] [HealthCheckService] WARN 
      io.camunda.zeebe.broker.system - Partition-3 failed, marking it as unhealthy: Partition-3{status=UNHEALTHY, issue=HealthIssue[message=null, throwable=null, cause=ZeebePartition-3{status=UNHEALTHY, issue=HealthIssue[message=Services not installed, throwable=null, cause=null]}]}
2024-01-30 14:23:26.321 [Broker-1] [zb-actors-0] [ZeebePartition-3] INFO 
      org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@2079f6cf)), function-provider: io.camunda.zeebe.feel.impl.FeelFunctionProvider@7cbc1e70, clock: io.camunda.zeebe.engine.processing.bpmn.clock.ZeebeFeelEngineClock@3beacd47, configuration: {externalFunctionsEnabled: false}]
2024-01-30 14:23:26.324 [Broker-1] [zb-actors-0] [ZeebePartition-3] INFO 
      org.camunda.dmn.DmnEngine - DMN-Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@374d1550)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@d87b90c, audit-loggers: List(), configuration: Configuration(false,false,false)]
2024-01-30 14:23:26.324 [Broker-1] [zb-actors-0] [ZeebePartition-3] INFO 
      org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(org.camunda.dmn.NoUnpackValueMapper@184bab7a)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@d87b90c, clock: SystemClock, configuration: {externalFunctionsEnabled: false}]
2024-01-30 14:23:26.326 [Broker-1] [zb-actors-0] [ZeebePartition-3] INFO 
      org.camunda.dmn.DmnEngine - DMN-Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.feel.impl.MessagePackValueMapper@49148916)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@d87b90c, audit-loggers: List(), configuration: Configuration(false,false,false)]
2024-01-30 14:23:26.326 [Broker-1] [zb-actors-0] [ZeebePartition-3] INFO 
      org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(org.camunda.dmn.NoUnpackValueMapper@d49cdda)), function-provider: org.camunda.feel.context.FunctionProvider$EmptyFunctionProvider$@d87b90c, clock: SystemClock, configuration: {externalFunctionsEnabled: false}]
2024-01-30 14:23:26.363 [Broker-1] [zb-actors-0] [ZeebePartition-3] ERROR
      io.camunda.zeebe.broker.system - Failed to install partition 3
java.lang.IndexOutOfBoundsException: index=30 length=8 capacity=30
	at org.agrona.AbstractMutableDirectBuffer.boundsCheck0(AbstractMutableDirectBuffer.java:1719) ~[agrona-1.20.0.jar:1.20.0]
	at org.agrona.AbstractMutableDirectBuffer.getLong(AbstractMutableDirectBuffer.java:129) ~[agrona-1.20.0.jar:1.20.0]
	at io.camunda.zeebe.db.impl.DbLong.wrap(DbLong.java:27) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.DbCompositeKey.wrap(DbCompositeKey.java:47) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.visit(TransactionalColumnFamily.java:450) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$forEachInPrefix$20(TransactionalColumnFamily.java:387) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ColumnFamilyContext.withPrefixKey(ColumnFamilyContext.java:112) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.forEachInPrefix(TransactionalColumnFamily.java:371) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.forEachInPrefix(TransactionalColumnFamily.java:342) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$forEach$7(TransactionalColumnFamily.java:162) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$ensureInOpenTransaction$19(TransactionalColumnFamily.java:325) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction(DefaultTransactionContext.java:31) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.ensureInOpenTransaction(TransactionalColumnFamily.java:324) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.forEach(TransactionalColumnFamily.java:160) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.to_8_4.DbSignalSubscriptionMigrationState.migrateSignalSubscriptionStateForMultiTenancy(DbSignalSubscriptionMigrationState.java:44) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.DbMigrationState.migrateSignalSubscriptionStateForMultiTenancy(DbMigrationState.java:458) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.to_8_4.MultiTenancySignalSubscriptionStateMigration.runMigration(MultiTenancySignalSubscriptionStateMigration.java:30) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.DbMigratorImpl.runMigration(DbMigratorImpl.java:170) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.DbMigratorImpl.handleMigrationTask(DbMigratorImpl.java:143) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.DbMigratorImpl.lambda$runMigrations$1(DbMigratorImpl.java:93) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInNewTransaction(DefaultTransactionContext.java:61) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction(DefaultTransactionContext.java:33) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.migration.DbMigratorImpl.runMigrations(DbMigratorImpl.java:91) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.broker.system.partitions.impl.steps.MigrationTransitionStep.transitionTo(MigrationTransitionStep.java:52) ~[zeebe-broker-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.broker.system.partitions.impl.PartitionTransitionProcess.lambda$proceedWithTransition$1(PartitionTransitionProcess.java:94) ~[zeebe-broker-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:120) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.4.0.jar:8.4.0]
2024-01-30 14:23:26.366 [Broker-1] [zb-actors-0] [ZeebePartition-3] WARN 
      io.camunda.zeebe.broker.system - ZeebePartition-3 failed, marking it as unhealthy: ZeebePartition-3{status=UNHEALTHY, issue=HealthIssue[message=Transition from null on term 0 appears blocked, step Migration has been running for PT0S, throwable=null, cause=null]}
2024-01-30 14:23:26.367 [Broker-1] [zb-actors-0] [HealthCheckService] WARN 
      io.camunda.zeebe.broker.system - Partition-3 failed, marking it as unhealthy: Partition-3{status=UNHEALTHY, issue=HealthIssue[message=null, throwable=null, cause=ZeebePartition-3{status=UNHEALTHY, issue=HealthIssue[message=Transition from null on term 0 appears blocked, step Migration has been running for PT0S, throwable=null, cause=null]}]}
```

</p>
</details>

that than are followed by a ton of:
<details><summary>IllegalStateException</summary>
<p>

```
2024-01-30 14:23:28.793 [Broker-1] [zb-actors-0] [] WARN 
      io.camunda.zeebe.topology.gossip.ClusterTopologyGossiper - Failed to sync with process-control-tower-gateway-deployment-5f757c8784-clk8r
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Unknown status: CHANGE_STATUS_UNKNOWN
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$23(NettyMessagingService.java:607) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31) ~[guava-33.0.0-jre.jar:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$26(NettyMessagingService.java:605) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
	at io.atomix.cluster.messaging.impl.AbstractClientConnection.dispatch(AbstractClientConnection.java:43) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.atomix.cluster.messaging.impl.AbstractClientConnection.dispatch(AbstractClientConnection.java:30) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.atomix.cluster.messaging.impl.NettyMessagingService$MessageDispatcher.channelRead0(NettyMessagingService.java:1109) ~[zeebe-atomix-cluster-8.4.0.jar:8.4.0]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407) ~[netty-transport-classes-epoll-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.104.Final.jar:4.1.104.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.104.Final.jar:4.1.104.Final]
	at java.base/java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: java.lang.IllegalStateException: Unknown status: CHANGE_STATUS_UNKNOWN
	at io.camunda.zeebe.topology.serializer.ProtoBufSerializer.toChangeStatus(ProtoBufSerializer.java:677) ~[zeebe-cluster-topology-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.topology.serializer.ProtoBufSerializer.decodeCompletedChange(ProtoBufSerializer.java:342) ~[zeebe-cluster-topology-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.topology.serializer.ProtoBufSerializer.decodeClusterTopology(ProtoBufSerializer.java:109) ~[zeebe-cluster-topology-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.topology.serializer.ProtoBufSerializer.decode(ProtoBufSerializer.java:78) ~[zeebe-cluster-topology-8.4.0.jar:8.4.0]
	... 32 more
```

</p>
</details>

**Environment:**
- OS: Linux (`camunda/broker` image)
- Zeebe Version: 8.4.0
- Configuration:
  ```
            - name: ZEEBE_BROKER_CLUSTER_CLUSTERNAME
              value: "process-control-tower-zeebe"
            - name: ZEEBE_GATEWAY_CONTACT_POINT
              value: "process-control-tower-gateway-service"
            - name: ZEEBE_LOG_LEVEL
              value: "warn"
            - name: ZEEBE_BROKER_CLUSTER_PARTITIONSCOUNT
              value: "3"
            - name: ZEEBE_BROKER_CLUSTER_CLUSTERSIZE
              value: "3"
            - name: ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR
              value: "3"
            - name: ZEEBE_BROKER_THREADS_CPUTHREADCOUNT
              value: "2"
            - name: ZEEBE_BROKER_THREADS_IOTHREADCOUNT
              value: "2"
            - name: ZEEBE_BROKER_GATEWAY_ENABLE
              value: "false"
            - name: ZEEBE_BROKER_EXPORTERS_HAZELCAST_JARPATH
              value: exporters/zeebe-hazelcast-exporter.jar
            - name: ZEEBE_BROKER_EXPORTERS_HAZELCAST_CLASSNAME
              value: io.zeebe.hazelcast.exporter.HazelcastExporter
            - name: ZEEBE_BROKER_NETWORK_COMMANDAPI_PORT
              value: "26501"
            - name: ZEEBE_BROKER_NETWORK_INTERNALAPI_PORT
              value: "26502"
            - name: ZEEBE_BROKER_NETWORK_MONITORINGAPI_PORT
              value: "9600"
            - name: ZEEBE_BROKER_DATA_DISK_FREE_SPACE_REPLICATION
              value: "1408MB" # default 1024MB. See https://docs.camunda.io/docs/self-managed/zeebe-deployment/operations/disk-space/
            - name: ZEEBE_BROKER_DATA_DISK_FREE_SPACE_PROCESSING # default 2048MB
              value: "2048MB" # default 2048MB. See https://docs.camunda.io/docs/self-managed/zeebe-deployment/operations/disk-space/
            - name: ZEEBE_BROKER_EXECUTION_METRICS_EXPORTER_ENABLED
              value: "true" # export
            - name: K8S_POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: JAVA_TOOL_OPTIONS
              # align the max heap should be ca 80% of max mem of the container
              value: "-Xmx3500M -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/usr/local/zeebe/data -XX:ErrorFile=/usr/local/zeebe/data/zeebe_error%p.log -XX:+ExitOnOutOfMemoryError"
  ```

**Comment by oleschoenburg** [comment]
Thanks for reporting! The actual issue was the `IndexOutOfBounds` which is being tracked here: https://github.com/camunda/zeebe/issues/15804

**Comment by oleschoenburg** [comment]
Unfortunately there's no workaround for now, you'll have to work for https://github.com/camunda/zeebe/issues/15804 to be fixed and released in 8.4. This will probably be 8.4.2.

**Comment by michaelholtermann** [comment]
Thanks a lot. My bad, I did not found that issue.

I'll track #15804 then.

**Comment by michaelholtermann** [comment]
FTR: #15804 solved my issue, I now upgrade to `8.4.2`.
